Serge/node/exercices/class.js

```diff
@@ -0,0 +1,127 @@
+        } else {
+            console.log('+ Le Joueur attaque et inflige ' + degats + ' points de dégâts !');
+        }
+        return degats;
+    };
+    subieDesDegats(degats) {
+        super.subieDesDegats(degats);
+        if (this.pointsDeVie <= 0) {
+            console.log('  - RIP... Le joueur reçoit ' + degats + ' points de dégâts... Il est mort !');
+            console.log('+++ Le joueur s\'est battu vaillament et a tué ' + score + ' bestioles... Gloire à lui !! +++');
+        } else if (this.pointsDeVie <= 10 && this.pointsDeVie > 1) {
+            console.log('  - Le joueur n\'a plus que ' + this.pointsDeVie + ' points de vie... La fin approche !')
+        } else if (this.pointsDeVie === 1) {
+            console.log('  - Le joueur n\'a plus que ' + this.pointsDeVie + ' point de vie... Sa vie ne tient qu\'à un fil !')
+        } else if (degats === 6) {
+            console.log('  - Le joueur encaisse difficilement l\'attaque en mode BERSERK de ' + degats + ' points de dégâts... Il accuse le coup !!')
+        } else {
+            console.log('  - Le joueur reçoit ' + degats + ' points de dégâts. Il lui reste ' + this.pointsDeVie + ' points de vie...');
+        }
+    };
+};
+
+class Monster extends Personnage {
+    constructor() {
+        super(20, 1, 6)
+    };
+    Attaque() {
+        const degats = super.Attaque();
+        if (degats === 6) {
+            console.log('+ AARGHH !!! La bestiole attaque en mode BERSERK et inflige ' + degats + ' points de dégâts !!!');
+        } else {
+            console.log('+ La Bestiole attaque à son tour et inflige ' + degats + ' points de dégâts !');
+        }
+        return degats;
+    };
+    subieDesDegats(degats) {
+        super.subieDesDegats(degats);
+        if (this.pointsDeVie <= 0) {
+            console.log('  - La bestiole est morte !');
+            score = score + 1;
+        } else {
+            console.log('  - La bestiole reçoit ' + degats + ' points de dégâts. Il lui reste ' + this.pointsDeVie + ' points de vie...');
+        }
+    };
+};
+
+class ChildMonster extends Personnage {
+    constructor() {
+        super(10, 0, 4)
+    };
+    Attaque() {
+        const degats = super.Attaque();
+        console.log('+ La petite attaque à son tour Bestiole inflige ' + degats + ' points de dégâts !');
+        return degats;
+    };
+    subieDesDegats(degats) {
+        super.subieDesDegats(degats);
+        if (this.pointsDeVie <= 0) {
+            console.log('  - La petite bestiole est morte !');
+            score = score + 1;
+        } else {
+            console.log('  - La petite bestiole reçoit ' + degats + ' points de dégâts. Il lui reste ' + this.pointsDeVie + ' points de vie...');
+        }
+    };
+};
+
+let score = 0;
+const joueur = new Player();
+
+function joueurVersusBestiole() {
+    const bestiole = new Monster();
```

Comment:
C'est ici que tu peux utiliser le modulo voici un exemple : https://stackoverflow.com/questions/16505559/how-can-i-use-modulo-operator-in-javascript

Rappel : Tous les 3 monstres je veux un petit monstre